File: project/BBWT.Client/src/app/main/formio/multi-user-form/dto/multi-user-form-associations.dto.ts
import { FormIOData, FormIODefinition } from "@features/bb-formio";
import { MultiUserFormAssociationLink, NewMultiUserFormAssociationLink } from "./multi-user-form-association-links.dto";
import { MultiUserFormDef } from "./multi-user-form.dto";
import { FormRevisionDTO } from "@features/bb-formio/dto/form-revision";

export class MultiUserFormAssociation {
    id?: string;
    id_original?: number;
    description: string;
    formData: FormIOData;
    formDataId: string | number;
    multiUserFormDefinitionId: number | string;
    multiUserFormDefinition: MultiUserFormDef;
    created: Date;
    multiUserFormAssociationLinks: MultiUserFormAssociationLink[];
    activeStageAssociation: number[];
    activeStepSequenceIndex: number;
    totalSequenceSteps: number;

    formDefinition: FormIODefinition;
    formRevision: FormRevisionDTO;
}

export class NewMultiUserFormAssociation {
    description: string;
    multiUserFormDefinitionId: number;
    created: Date;
    multiUserFormAssociationLinks: NewMultiUserFormAssociationLink[];
}

File: modules/BBWM.Formio/DTO/MultiUserFormAssociationsDTO.cs
using BBWM.Core.DTO;
using BBWM.FormIO.Models;

namespace BBWM.FormIO.DTO
{
    /// <summary>
    /// This entity is the one that links a MUF definition with its Form Data for each Stage and allow a way to track which stage are completed or not.
    /// </summary>
    public class MultiUserFormAssociationsDTO : IDTO
    {
        public int Id { get; set; }
        public DateTime Created { get; set; }
        public string Description { get; set; } = null!;
        public int TotalSequenceSteps { get; set; }
        public int ActiveStepSequenceIndex { get; set; }

        // Foreign keys and navigational properties.
        public int FormDataId { get; set; }
        public virtual FormData FormData { get; set; } = null!;

        public int MultiUserFormDefinitionId { get; set; }
        public virtual MultiUserFormDefinition MultiUserFormDefinition { get; set; } = null!;

        public List<int> ActiveStageAssociation { get; set; } = new();

        public virtual FormRevisionDTO FormRevision { get; set; } = null!;
        public virtual FormDefinitionDTO FormDefinition { get; set; } = null!;

        public virtual ICollection<MultiUserFormAssociationLinks> MultiUserFormAssociationLinks { get; set; }
            = new List<MultiUserFormAssociationLinks>();
    }

    public class NewMultiUserFormAssociationsDTO
    {
        public string Description { get; set; } = null!;
        public DateTime Created { get; set; }

        // Foreign keys and navigational properties.
        public int MultiUserFormDefinitionId { get; set; }
        public List<NewMultiUserFormAssociationLinksDTO> MultiUserFormAssociationLinks { get; set; } = new();
    }
}